build-tools/Xamarin.Android.Tools.BootstrapTasks/Xamarin.Android.Tools.BootstrapTasks/WaitForAndroidEmulator.cs

```diff
@@ -0,0 +1,47 @@
+using System;
+using System.Collections.Generic;
+using System.Threading;
+
+namespace Xamarin.Android.Tools.BootstrapTasks
+{
+	public class WaitForAndroidEmulator : Adb
+	{
+		int bootCompleted = -1;
+
+		public override bool Execute ()
+		{
+			var endTime = DateTime.UtcNow.AddMilliseconds (Timeout);
+			while (DateTime.UtcNow < endTime && bootCompleted != 1) {
+				base.Execute ();
+				Thread.Sleep (3000);
+			}
+
+			if (bootCompleted != 1) {
+				Log.LogError ($"Emulator '{AdbTarget}' did not finish launching in {Timeout} ms.");
+				return false;
+			} else {
+				return true;
+			}
```

Comment:
We usually have the patter of `Execute()` doing: `return !Log.HasLoggedErrors;`

Reply:
Thanks, updated